code/core/system/proxy-imu/include/Pololualtimu10device.h

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (C) 2015 Chalmers REVERE
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+#ifndef PROXY_POLOLUALTIMU10DEVICE_H
+#define PROXY_POLOLUALTIMU10DEVICE_H
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+
+#include "odvdimu/GeneratedHeaders_ODVDIMU.h"
+
+namespace opendlv {
+namespace core {
+namespace system {
+namespace proxy {
+
+class PololuAltImu10Device {
+   public:
+    PololuAltImu10Device(std::string const &, std::string &, bool &);
+
+    PololuAltImu10Device(PololuAltImu10Device const &) = delete;
+
+    PololuAltImu10Device &operator=(PololuAltImu10Device const &) = delete;
+
+    virtual ~PololuAltImu10Device();
+
+    void saveCalibrationFile();
+
+    opendlv::proxy::AccelerometerReading ReadAccelerometer();
+
+    opendlv::proxy::AltimeterReading ReadAltimeter();
+
+    opendlv::proxy::TemperatureReading ReadTemperature();
+
+    opendlv::proxy::CompassReading ReadCompass();
+
+    opendlv::proxy::GyroscopeReading ReadGyroscope();
+
+    bool IsInitialized() const;
+
+
+   private:
+    void I2cWriteRegister(uint8_t, uint8_t);
+    
+    bool loadCalibrationFile();
+
+    void accessLSM6();
+
+    void accessLIS3();
+
+    void accessLPS25();
+
+    void initLSM6();
+
+    void initLIS3();
+
+    void initLPS25();
+
+    void CalibrateCompass(float*);
+
+    int16_t m_deviceFile;
+
+    std::string m_calibrationFile;
+
+    std::string m_calibrationPath;
+
+    float m_compassMaxVal[3];
+
+    float m_compassMinVal[3];
+
+    float m_heavyAcc[3];
+
+    bool m_initialized;
+
+    bool m_debug;
+};
+
+namespace lis3RegAddr {
```

Comment:
I don't think we should use namespaces like this, instead put the name in the enum name. Also make sure to use an informative name.